### ISSUE TITLE
Using tuples for torch.max() 

### ISSUE THREAD

**Issue by SpaceMeerkat** [pytorch/pytorch#31042]
## 🚀 Feature
Introduce a torch.max() function that will allow the use of tuples when specifying the dimension over which to evaluate the maximum values in a tensor.

## Motivation

Numpy supports tuples, making this a simple one line problem:
`max_vals = array.max(axis=(1,2,3))`

Whereas, with PyTorch, one must use a more bloated approach:
`max_vals = tensor.max(1)[0].max(1)[0].max(1)[0]`

## Pitch

The fact we have to index the output of `torch.max()` operations to get the maximum values is actually a blessing in disguise as (for up to 2D inputs) you can see what indices the maximum values are located at. If we want to preserve this feature then perhaps an alternative function called torch.maxNDim() could be created for the integration of tuple dimensions without the functionality of returning the indices of those maximum values

## Alternatives

Alternatively, updating the current `torch.max()` function such that the second returned tensor gives an ND set of indices for the location of maximum values in an ND tensor would be fantastic. e.g. for getting the indices of the maximum values in a 3D tensor along the third axis, one can imagine you would return a 2D array of rows and column indices.

## Additional context

Here is some example code to show the current issue more clearly:

```
### Numpy version of the problem

import numpy as np

a = np.random.uniform(0,1,(5,1,3,3))
max_vals = a.max(axis=(1,2,3)) ### Clear and concise

### PyTorch version of the problem (currently)

import torch 

at = torch.tensor(a)
max_vals = at.max(1)[0].max(1)[0].max(1)[0] ### Bloated 
```

Additionally, this functionality could be extended for common numerical operations such as `torch.min()`, `torch.std()`, etc.

**Comment by zasdfgbnm** [comment]
This feature is currently under development at https://github.com/pytorch/pytorch/pull/43092, but it will be named `torch.amax`.

**Comment by f0k** [comment]
@SpaceMeerkat: #43092 has been merged, so I think this can be closed (unless you also want indices).

**Comment by SpaceMeerkat** [comment]
@zasdfgbnm and @f0k, apologies for not seeing this _issue_ raised in #28213 (pretty new to using the _issue_ functionality here). Looks like it's all sorted now (like @f0k, I also don't require indices), thanks both!
